tests/integration/sync_test.go

```diff
@@ -0,0 +1,123 @@
+//go:build integration
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+var DECK = filepath.Join("../../main.go")
+
+func getKongAddress() string {
+	address := os.Getenv("DECK_KONG_ADDR")
+	if address != "" {
+		return address
+	}
+	return "http://localhost:8001"
+}
+
+// cleanUpEnv removes all existing entities from Kong.
+func cleanUpEnv() error {
+	cmd := exec.Command(
+		"go",
+		"run",
+		DECK,
+		"reset",
+		"--force",
```

Comment:
Why not simply call main() as a Go command?

Reply:
Similar to the first point from https://github.com/Kong/deck/pull/572#issuecomment-1023621268 earlier, there are two approaches: invoke the actual command as a user would or invoke functions directly from the tests. I'd recommend starting with the latter, since most of your complex testing should go that route to allow for debugger usage and properly-typed checks.

Command invocation tests are useful also, but more to test a minimal "does the final product work/are there any issues with the wrappers we use around the functions we call directly in integration tests?" question.

Reply:
+2 ^

Reply:
+3 ^

I think we should agree on a plan for this effort, and this PR was open with this intent. Does the following sounds reasonable?

Limited to the scope of this PR:
- general CI setup and utilities (changes in `.ci/` and `.github/` folders and `Makefile`), already present in this PR 
- Simple test invoking `syncMain` directly and test results via API (this would require a bit of changes, as `syncMain` is unexported and also today there is no way to inject config parameters into it, which are populated as part of the root CLI)

On another PR:
- add more test cases to the above, including more entities etc

On an even further PR:
- add "e2e" tests testing the CLI usage (what I was doing here at this time)

Reply:
Rework on this direction: https://github.com/Kong/deck/pull/572/commits/1ca2c4966498a2a39a07c87117bfa84ccc910f9d